app/src/main/java/com/sci/coffeeandroid/feature/auth/data/service/AuthNetworkServiceImpl.kt

```diff
@@ -0,0 +1,103 @@
+package com.sci.coffeeandroid.feature.auth.data.service
+
+import com.sci.coffeeandroid.feature.auth.data.model.request.LoginRequestModel
+import com.sci.coffeeandroid.feature.auth.data.model.request.RegisterRequestModel
+import com.sci.coffeeandroid.feature.auth.data.model.response.Data
+import com.sci.coffeeandroid.feature.auth.data.model.response.LoginResponseModel
+import com.sci.coffeeandroid.feature.auth.data.model.response.OTPResponse
+import com.sci.coffeeandroid.feature.auth.data.model.response.PasswordResetResponse
+import com.sci.coffeeandroid.feature.auth.data.model.response.RegisterResponse
+import io.ktor.client.HttpClient
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+class AuthNetworkServiceImpl(httpClient: HttpClient) : AuthNetworkService {
+    override suspend fun register(
+        username: String,
+        email: String,
+        phone: String,
+        password: String
+    ): Result<RegisterResponse> {
+        val request = RegisterRequestModel(
+            username = username,
+            email = email,
+            phoneNumber = phone,
+            password = password
+        )
+        withContext(Dispatchers.IO) {
+            Thread.sleep(2000)
```

Comment:
why thread pause ??

Reply:
I want to add delay to show loading when I click login button.
Is there another way sir?